cypress/integration/core-opensearch-dashboards/opensearch-dashboards/datasource-management-plugin/1_create_datasource.spec.js

```diff
@@ -0,0 +1,260 @@
+      ).should('exist');
+    });
+
+    describe('Datasource can be created successfully', () => {
+      // No auth, with all required inputs, no optional description
+      it('with no auth and all required inputs', () => {
+        cy.get('[name="dataSourceTitle"]').type('test_noauth');
+        cy.get('[name="endpoint"]').type(endPointUrl);
+        cy.get('[for="no_auth"]').click();
+        cy.get('[type="submit"]').click();
+
+        cy.location('pathname', { timeout: 6000 }).should(
+          'include',
+          'app/management/opensearch-dashboards/dataSources'
+        );
+      });
+
+      //Basic auth, with all required inputs, no optional description
+      it('with basic auth and all required inputs', () => {
+        cy.get('[name="dataSourceTitle"]').type('test_auth');
+        cy.get('[name="endpoint"]').type(endPointUrl);
+        cy.get('[for="username_password"]').click();
+        cy.get('[data-test-subj="createDataSourceFormUsernameField"]').type(
+          username
+        );
+        cy.get('[data-test-subj="createDataSourceFormPasswordField"]').type(
+          password
+        );
+        cy.get('[type="submit"]').click();
+
+        cy.location('pathname', { timeout: 6000 }).should(
+          'include',
+          'app/management/opensearch-dashboards/dataSources'
+        );
+      });
+
+      // No auth, with all inputs
+      it('with no auth and all inputs', () => {
+        cy.get('[name="dataSourceTitle"]').type('test_noauth_with_all_Inputs');
+        cy.get('[name="dataSourceDescription"]').type(
+          'test if can create datasource with no auth successfully'
+        );
+        cy.get('[name="endpoint"]').type(endPointUrl);
+        cy.get('[for="no_auth"]').click();
+        cy.get('[type="submit"]').click();
+
+        cy.location('pathname', { timeout: 6000 }).should(
+          'include',
+          'app/management/opensearch-dashboards/dataSources'
+        );
+      });
+
+      // Basic auth, with all inputs
+      it('with basic auth and all inputs', () => {
+        cy.get('[name="dataSourceTitle"]').type('test_auth_with_all_Inputs');
+        cy.get('[name="dataSourceDescription"]').type(
+          'test if can create datasource with basic auth successfully'
+        );
+        cy.get('[name="endpoint"]').type(endPointUrl);
+        cy.get('[for="username_password"]').click();
+        cy.get('[data-test-subj="createDataSourceFormUsernameField"]').type(
+          username
+        );
+        cy.get('[data-test-subj="createDataSourceFormPasswordField"]').type(
+          password
+        );
+        cy.get('[type="submit"]').click();
+
+        cy.location('pathname', { timeout: 6000 }).should(
+          'include',
+          'app/management/opensearch-dashboards/dataSources'
+        );
+      });
+    });
+
+    //Title field validation
+    describe('Title validation', () => {
```

Comment:
the validation tests share a lot same of code, is it possible to apply Dynamic tests or create custom commands?
Please check out [the Dynamic tests examples from Cypress](https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/fundamentals__dynamic-tests) and [Index Management Plugin commands](https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress/utils/plugins/index-management-dashboards-plugin/commands.js)

Reply:
Thanks for sharing! I have looked into the documents that you provided. From my perspective, I think we can keep these code for now. Though the code looks like similar, but actually each field may have their own validation logic. For example, title need validate if there is duplicate datasource name but others does not need to do this validation. Also, endpoint URL has unique validation logic as well. The UX team may also change the design (eg. add inline error based on validation logic) in 2.5. Thus, I think keep these codes for now is a good way to clearly indicate each test step and each validation logic.